### ISSUE TITLE
dev sw file gets created at "base/base/dev-sw.js?dev-sw"

### ISSUE THREAD

**Issue by LWolf-roxtra** [vite-pwa/vite-plugin-pwa#683]
Having the following settings in vite.config.ts:

```
{
  base: "/my/app/",
  plugins: [
    VitePWA({
      stategies: "injectManifest",
      devOptions: {
        enabled: true,
        ...
     },
    ...
   })
 ]
}
```

Following the docs, i register the sw via this code:

```
const baseName = "/my/app/";
const serviceWorkerPath = import.meta.env.MODE === "production" ? `${baseName}sw.js` : `${baseName}dev-sw.js?dev-sw`;
const wb = new Workbox(serviceWorkerPath, {
    type: isProductionOrAppMode ? "classic" : "module",
  });
...
await wb.register();
```

In production stage, this registers the sw correctly, however in development, i get 404 error. After a lot of debugging and trying things out, i found out, that the sw file is being created at path "https://my-machine/my/app//my/app/dev-sw.js".

Knowing that, i found a workaround:
When setting the `serviceWorkerPath` constant like that, it works:
```const serviceWorkerPath = import.meta.env.MODE === "production" ? `${baseName}sw.js` : `${baseName}${baseName}dev-sw.js?dev-sw`;```
Conclusion: if I use the baseName constant twice, the path to the sw is correct, and the sw gets installed. 

I cannot find any other config parameter, where I have set the "/my/app/" base url/path, or seem to cannot find where I can set it, so it works.

Are you able to reproduce this aswell? Have I forgotten anything? Is this a bug maybe?

**Comment by userquin** [comment]
try:
- remove traling slash: `/my/app`
- include ìnjectRegister: false` in pwa options
- replace `const baseName = "/my/app/";` with `const baseName = import.meta.env.BASE_URL`
- add `/`  in the mapping

**Comment by LWolf-roxtra** [comment]
Thanks for your reply! 

Unfortunately:
- removing the trailing slash didn't help
- `injectRegister` is already set to `false`
- `baseName` actually is `import.meta.env.BASE_URL`

Do you have any other ideas on how to solve the problem or why it occurs?

**Comment by userquin** [comment]
Are you adding base in your html entry point?

**Comment by LWolf-roxtra** [comment]
> Are you adding base in your html entry point?

No, i am not

**Comment by userquin** [comment]
can you provide a minimal reproduction?

**Comment by LWolf-roxtra** [comment]
@userquin 
https://github.com/LWolf-roxtra/vite-plugin-pwa-issue-683/tree/main

Minimal reproduction here, just `git clone` the repo and follow `readme.md` instructions for running locally and reproducing the problem. Thank you for your help! :)

**Comment by userquin** [comment]
When resolving the dev sw url in dev plugin, vite removing the base from the id, and so the dev sw url is `/dev-sw.js?dev-sw` and it doesn't match `dev-sw.js?dev-sw`: the content will be empty and send with text/html content type:

![imagen](https://github.com/vite-pwa/vite-plugin-pwa/assets/6311119/5f3bbb04-e346-4cd9-a461-1c0a27427446)

**Comment by userquin** [comment]
@LWolf-roxtra should be fixed, released 0.19.5
